aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandler.java

```diff
@@ -76,21 +99,51 @@
 
     private GetPolicyResponse getExistingPolicy(AmazonWebServicesClientProxy proxy, String profilingGroupName) {
         return proxy.injectCredentialsAndInvokeV2(
-            GetPolicyRequest.builder().profilingGroupName(profilingGroupName).build(),
-            profilerClient::getPolicy
+                GetPolicyRequest.builder().profilingGroupName(profilingGroupName).build(),
+                profilerClient::getPolicy
         );
     }
 
+    private void updateNotificationChannels(String pgName, AmazonWebServicesClientProxy proxy, List<Channel> currentChannels) {
+        List<Channel> existingChannels = getExistingNotificationConfiguration(proxy, pgName).notificationConfiguration().channels();
+        updateNotificationChannels(existingChannels, currentChannels, pgName, proxy);
+    }
+
+    private GetNotificationConfigurationResponse getExistingNotificationConfiguration(AmazonWebServicesClientProxy proxy, String profilingGroupName) {
+        return proxy.injectCredentialsAndInvokeV2(GetNotificationConfigurationRequest.builder()
+                        .profilingGroupName(profilingGroupName)
+                        .build(),
+                profilerClient::getNotificationConfiguration);
+    }
+
+    // Generate the change-set and add / delete notification channel based on change-set
+    private void updateNotificationChannels(List<Channel> existingChannels, List<Channel> currentChannels, String pgName, AmazonWebServicesClientProxy proxy) {
+        Set<String> existingChannelsSet = existingChannels.stream().map(Channel::uri).collect(Collectors.toSet());
+        Set<String> currentChannelsSet = currentChannels.stream().map(Channel::uri).collect(Collectors.toSet());
+        for (Channel channel : existingChannels) {
+            if (!currentChannelsSet.contains(channel.uri())) {
+                // Can assert that channel.id() exists here, since this is from an existingChannel, which is fetched via the GetNotificationConfiguration, which has the id attached
+                NotificationChannelHelper.deleteNotificationChannel(pgName, channel.id(), proxy, profilerClient);
+            }
```

Comment:
What happens when the customer kept the channel, but changes its id? I think we should either apply the new id, or else reject the new configuration. Silently ignoring it witll probably cause confusion.